src/api/java/baritone/api/Settings.java

```diff
@@ -1385,14 +1421,82 @@ public final Type getType() {
         settingTypes = Collections.unmodifiableMap(tmpSettingTypes);
     }
 
+    /**
+     * Register a new setting.
+     *
+     * @param name         the name of the setting
+     * @param defaultValue the default value of the setting
+     * @param <T>          the type of the setting value
+     * @return the newly registered setting
+     */
+    public <T> Setting<T> registerSetting(String name, T defaultValue) {
+        Setting<T> setting = new Setting<>(defaultValue);
+        setting.name = name;
+        return superSecretRegisterSetting(setting, name.toLowerCase(), defaultValue.getClass());
+    }
+
+    /**
+     * Register settings from the fields of the given class.
+     *
+     * @param inputClass the class containing the settings fields
+     */
+    public void registerSettingsFromClass(Class<?> inputClass) {
+        Field[] fields = inputClass.getDeclaredFields();
+
+        for (Field field : fields) {
+            if (field.getType().equals(Setting.class)) {
+                try {
+                    field.setAccessible(true);
+                    Setting<?> setting = (Setting<?>) field.get(this);
+                    String name = field.getName();
+                    setting.name = name;
+                    String lowercaseName = name.toLowerCase();
+                    superSecretRegisterSetting(setting, lowercaseName, ((ParameterizedType) field.getGenericType()).getActualTypeArguments()[0]);
+                } catch (IllegalAccessException e) {
+                    throw new IllegalStateException("Error accessing setting field", e);
+                }
+            }
+        }
+    }
+
+    /**
+     * Retrieves a list of settings of a specific type.
+     *
+     * @param <T>   the type of the settings to retrieve
+     * @param clazz the class representing the type of settings
+     * @return a list of settings of the specified type
+     */
     @SuppressWarnings("unchecked")
-    public <T> List<Setting<T>> getAllValuesByType(Class<T> cla$$) {
+    public <T> List<Setting<T>> getAllValuesByType(Class<T> clazz) {
```

Comment:
Please revert this (See https://github.com/cabaletta/baritone/commit/a7add6bc13df16cfbe42e6ae02ff1c6a0c4c1313)